### ISSUE TITLE
Regression: `loadPackage` doesn't return entries for wheel URLs

### ISSUE THREAD

**Issue by hoodmane** [pyodide/pyodide#4330]
## 🐛 Bug

In Pyodide stable we see the following:
```
$ await pyodide.loadPackage("https://cdn.jsdelivr.net/pyodide/v0.24.1/full/multidict-6.0.4-cp311-cp311-emscripten_3_1_45_wasm32.whl");
Loading multidict
Loaded multidict
undefined
```

I think this got broken in #3843. On tip of tree we see:
```
$ await pyodide.loadPackage("https://cdn.jsdelivr.net/pyodide/dev/full/multidict-6.0.4-cp311-cp311-emscripten_3_1_45_wasm32.whl");
Loading multidict
[ ]
```
It still successfully loads the wheel but
1. it doesn't print `"Loaded multidict"`
2. it returns an empty list, whereas I'd expect it to return something like
```
[{ name: "multidict", version: "???", fileName: "multidict-6.0.4-cp311-cp311-emscripten_3_1_45_wasm32.whl", packageType: "package" }]
```

cc @ryanking13 @RulerOfCakes

**Comment by ryanking13** [comment]
Looks like this line is causing the problem:

https://github.com/pyodide/pyodide/blob/9e90f695ab9fe962a96a781c02cddbea550601d3/src/js/load-package.ts#L379-L380

since packages from URL does not have metadata in the pyodide-lock.json

**Comment by RulerOfCakes** [comment]
Oops, I should've kept the `"Loaded ..."` message for this case as well.. I'll try to come up with a fix for both problems.

**Comment by hoodmane** [comment]
This also presumably means that test coverage for `loadPackage` is insufficient.

**Comment by ryanking13** [comment]
Closing as resolved by #4336
